bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/InitialFileManager.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Bachmann electronic GmbH and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp;
```

Comment:
Hm, but that 's what I had in mind as I created it. It provides the getter an ssetter for the inital URI. This class must be a singleton, since the lsp and clangd bundle bundles should access it via the LspPlugin class.

Reply:
I can suggest my changes after merging yours